### PR TITLE
fix!: persist other features when changing the community feature

### PR DESCRIPTION
**Pull request by onerandomusername**
## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/guild.py [hunk 1]
```diff
@@ -1974,16 +1974,36 @@ async def edit(
...
+            # The issue is removing community also removes partner status, and discord is adding the ability to
+            # enable or disable invite usage for guilds, which is problematically triggered by a feature.
+            # To enable or disable that feature, we will need to provide all of the existing features in advance.
+            # That is problematic to say the least. By reimplementing this list to only provide the features if we make a change
+            # we minimize the possible times that we will provide features when no changes are made, and possibly affect other features.
+            if self.unavailable:
+                raise RuntimeError(
+                    "cannot modify features of an unavailable guild due to potentially destructive results."
+                )
+            features = self.features
```
Comment (shiftinv) [inline disnake/guild.py:1988]: this mutates the instance's `features` list, should either copy or make it a set which would also simplify some code below

File: disnake/guild.py [hunk 2]
```diff
@@ -1974,16 +1974,36 @@ async def edit(
...
                     )
+            else:
+                try:
+                    features.remove("COMMUNITY")
+                except ValueError:
+                    features = MISSING
 
-            fields["features"] = features
+            if features is not MISSING:
+                fields["features"] = features
```
Comment (shiftinv) [inline disnake/guild.py:2006]: why set features to `MISSING` and handle it here? sending the unmodified features list should work just fine.

**Review by shiftinv — CHANGES_REQUESTED** [review]

File: disnake/guild.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/guild.py]: https://github.com/DisnakeDev/disnake/pull/705/commits/badbf3f29b225c6bb274f88a910581c4f6c15eb8

File: disnake/guild.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/guild.py]: https://github.com/DisnakeDev/disnake/pull/705/commits/badbf3f29b225c6bb274f88a910581c4f6c15eb8

File: disnake/guild.py [hunk 3]
```diff
@@ -1975,16 +1975,32 @@ async def edit(
...
-            if community:
-                if "rules_channel_id" in fields and "public_updates_channel_id" in fields:
-                    features.append("COMMUNITY")
+            # if we don't have complete feature information for the guild
+            # it is possible to disable or enable other features that we didn't intend to touch
+            # The issue is removing community also removes partner status, and discord is adding the ability to
+            # enable or disable invite usage for guilds, which is problematically triggered by a feature.
+            # To enable or disable that feature, we will need to provide all of the existing features in advance.
+            # That is problematic to say the least. By reimplementing this list to only provide the features if we make a change
+            # we minimize the possible times that we will provide features when no changes are made, and possibly affect other features.
```
Comment (shiftinv) [inline disnake/guild.py:1984]:
```suggestion
            # If we don't have complete feature information for the guild,
            # it is possible to disable or enable other features that we didn't intend to touch.
            # To enable or disable a feature, we will need to provide all of the existing features in advance.
```

**Review by shiftinv — APPROVED** [review]
other than that one comment, lgtm

**Pull request merged**